pkg/controller/monitor/monitor_controller_test.go

```diff
@@ -0,0 +1,126 @@
+		Expect(appsv1.SchemeBuilder.AddToScheme(scheme)).NotTo(HaveOccurred())
+		Expect(rbacv1.SchemeBuilder.AddToScheme(scheme)).NotTo(HaveOccurred())
+
+		// Create a client that will have a crud interface of k8s objects.
+		ctx = context.Background()
+		cli = fake.NewClientBuilder().WithScheme(scheme).Build()
+
+		// Create an object we can use throughout the test to do the compliance reconcile loops.
+		mockStatus = &status.MockStatus{}
+		mockStatus.On("AddCronJobs", mock.Anything)
+		mockStatus.On("AddDaemonsets", mock.Anything)
+		mockStatus.On("AddDeployments", mock.Anything).Return()
+		mockStatus.On("AddStatefulSets", mock.Anything)
+		mockStatus.On("ClearDegraded")
+		mockStatus.On("IsAvailable").Return(true)
+		mockStatus.On("OnCRFound").Return()
+		mockStatus.On("ReadyToMonitor")
+
+		// Create an object we can use throughout the test to do the compliance reconcile loops.
+		r = ReconcileMonitor{
+			client:   cli,
+			scheme:   scheme,
+			provider: operatorv1.ProviderNone,
+			status:   mockStatus,
+		}
+
+		// We start off with a 'standard' installation, with nothing special
+		Expect(cli.Create(ctx, &operatorv1.Installation{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "default",
+			},
+			Status: operatorv1.InstallationStatus{
+				Variant:  operatorv1.TigeraSecureEnterprise,
+				Computed: &operatorv1.InstallationSpec{},
+			},
+			Spec: operatorv1.InstallationSpec{
+				Variant:  operatorv1.TigeraSecureEnterprise,
+				Registry: "some.registry.org/",
+			},
+		})).To(BeNil())
+
+		// Apply the Monitor CR to the fake cluster.
+		Expect(cli.Create(ctx, &operatorv1.Monitor{
+			TypeMeta:   metav1.TypeMeta{Kind: "Monitor", APIVersion: "operator.tigera.io/v1"},
+			ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+		})).NotTo(HaveOccurred())
+	})
+
+	Context("image reconciliation", func() {
+		It("should create Prometheus related resources", func() {
+			am := &monitoringv1.Alertmanager{}
+			Expect(cli.Get(ctx, client.ObjectKey{Name: render.CalicoNodeAlertmanager, Namespace: common.TigeraPrometheusNamespace}, am)).To(HaveOccurred())
+			p := &monitoringv1.Prometheus{}
+			Expect(cli.Get(ctx, client.ObjectKey{Name: render.CalicoNodePrometheus, Namespace: common.TigeraPrometheusNamespace}, p)).To(HaveOccurred())
+			pm := &monitoringv1.PodMonitor{}
+			Expect(cli.Get(ctx, client.ObjectKey{Name: render.FluentdMetrics, Namespace: common.TigeraPrometheusNamespace}, pm)).To(HaveOccurred())
+			pr := &monitoringv1.PrometheusRule{}
+			Expect(cli.Get(ctx, client.ObjectKey{Name: render.TigeraPrometheusDPRate, Namespace: common.TigeraPrometheusNamespace}, pr)).To(HaveOccurred())
+			sm := &monitoringv1.ServiceMonitor{}
+			Expect(cli.Get(ctx, client.ObjectKey{Name: render.CalicoNodeMonitor, Namespace: common.TigeraPrometheusNamespace}, sm)).To(HaveOccurred())
+			Expect(cli.Get(ctx, client.ObjectKey{Name: render.ElasticsearchMetrics, Namespace: common.TigeraPrometheusNamespace}, sm)).To(HaveOccurred())
```

Comment:
If you want to first check that they don't exist then I think you should add that to the BeforeEach. Maybe add a comment before them also please. I was confused for a bit until I realized you were checking that they didn't exist first.
Also checking they don't exist is a prerequisite and not what we're trying to test.

Reply:
This UT is updated as suggested.